ons_alpha/bulletins/models.py

```diff
@@ -121,6 +124,14 @@ def get_context(self, request, *args, **kwargs):
         context["toc"] = self.toc
         return context
 
+    def save(self, *args, **kwargs):
+        # Set the previous_version field for corrections
+        if self.revisions.exists() and not self.previous_version:
+            latest_revision = self.revisions.order_by("-created_at").first()
```

Comment:
Question: Does this handle revisions which aren't published yet?

Reply:
@RealOrangeOne  - Thank you for the feedback. The necessary changes to the block definition have been made to ensure the previous_version field is correctly stored in the CorrectionBlock. The migration file now includes the appropriate changes to reflect this. Additionally, I've used Django's content types to validate that page_revision is a revision for the current page in the database.

Reply:
The latest created revision isn't necessarily the one which was previously published, so I don't think this I'm afraid.